cg/test/subjects/data-shares.api.spec.js

```diff
@@ -0,0 +1,140 @@
+jest.mock('../../src/utils/blockchain/web3-provider-factory');
+const { initResources, fetch, closeResources } = require('../utils');
+const { db } = require('../../src/db');
+const { subjectJWT } = require('../../src/utils/jwt');
+const { deployContract } = require('../blockchain-setup');
+const { hash } = require('../../src/utils/encryption');
+const { BadRequest } = require('../../src/utils/errors');
+
+beforeAll(initResources);
+
+afterAll(closeResources);
+
+async function createUser(id) {
+  const token = await subjectJWT.sign({ subjectId: id });
+  await fetch('/api/subject/give-consent', {
+    method: 'POST',
+    body: { personalData: { name: 'test' } },
+    headers: {
+      Authorization: `Bearer ${token}`
+    }
+  });
+  return token;
+}
+
+function fetchWithAuthorization(path, token, overrides) {
+  return fetch(
+    path,
+    Object.assign(
+      {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${token}`
+        }
+      },
+      overrides
+    )
+  );
+}
+beforeAll(async () => {
+  try {
+    await deployContract();
+  } catch (e) {
+    console.error(`Failed deploying contract ${e.toString()}`);
+  }
+  await initResources();
+});
+
+describe('Listing data-shares', () => {
```

Comment:
I'd add a test case for listing data shares that returns an empty array – jut to be sure there's no HTTP 500